spectractor/config.py

```diff
@@ -129,7 +121,26 @@ def load_config(config_filename, rebin=True):
                 print(f"x {options}: {value}\t=> parameters.{options.upper()}: {par}\t {type(par)}")
 
 
-def apply_rebinning_to_parameters():
+def update_derived_parameters():
+    # Derive other parameters
+    parameters.CALIB_BGD_NPARAMS = parameters.CALIB_BGD_ORDER + 1
+    parameters.LAMBDAS = np.arange(parameters.LAMBDA_MIN, parameters.LAMBDA_MAX, 1)
+    parameters.MY_FORMAT = "%(asctime)-20s %(name)-10s %(funcName)-20s %(levelname)-6s %(message)s"
+    logging.basicConfig(format=parameters.MY_FORMAT, level=logging.WARNING)
+    parameters.CCD_ARCSEC2RADIANS = np.pi / (180. * 3600.)  # conversion factor from arcsec to radians
+    parameters.OBS_DIAMETER = parameters.OBS_DIAMETER * units.m  # Diameter of the telescope
+    parameters.OBS_SURFACE = np.pi * parameters.OBS_DIAMETER ** 2 / 4.  # Surface of telescope
+    # Conversion factor
+    # Units of SEDs in flam (erg/s/cm2/nm) :
+    parameters.hc = const.h * const.c  # h.c product of fontamental constants c and h
+    parameters.SED_UNIT = 1 * units.erg / units.s / units.cm ** 2 / units.nanometer
+    parameters.TIME_UNIT = 1 * units.s  # flux for 1 second
+    parameters.wl_dwl_unit = units.nanometer ** 2  # lambda.dlambda  in wavelength in nm
+    parameters.FLAM_TO_ADURATE = ((parameters.OBS_SURFACE * parameters.SED_UNIT * parameters.TIME_UNIT
+                                   * parameters.wl_dwl_unit / parameters.hc / parameters.CCD_GAIN).decompose()).value
+
+
+def apply_rebinning_to_parameters(reverse=False):
```

Comment:
yes I tested it and force the types of the parameters inside the functions.

Reply:
Hmmm I see your point now. Indeed for odd numbers it can fail. Then I remove it, and in particular in the SpectrumSimulation class where it was used. Maybe now with the parameters saved in the Spectrum output it is not necesssary.

Reply:
I think removing it is the right answer, mainly because it's not just odd numbers - everything that isn't a power of two will behave like an odd number with a high enough rebinning factor, you just see it with rebin of 2 for odd numbers, but for example, 30 doesn't round-trip correctly with a rebin of 4 applied.